### PR TITLE
storage/mvcc: add unit test cases

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR adds some unit test cases for storage/mvcc.

PTAL @AndreMouche @disksing @siddontang

### CODE REVIEW

**Comment by disksing** [comment]
LGTM. PTAL @zhangjinpeng1987

File: src/storage/mvcc/lock.rs [hunk 1]
```diff
@@ -130,3 +130,67 @@ impl Lock {
...
+    #[test]
+    fn test_lock_type() {
+        let (key, value) = (b"key", b"value");
+        let mut tests =
+            vec![(Mutation::Put((make_key(key), value.to_vec())), LockType::Put, FLAG_PUT),
+                 (Mutation::Delete(make_key(key)), LockType::Delete, FLAG_DELETE),
+                 (Mutation::Lock(make_key(key)), LockType::Lock, FLAG_LOCK)];
+        for (i, (mutation, lock_type, flag)) in tests.drain(..).enumerate() {
+            let lt = LockType::from_mutation(&mutation);
+            if lt != lock_type {
```
Comment (siddontang) [inline src/storage/mvcc/lock.rs]: `assert(lt != lock_type, "#{}...", i...) `

File: src/storage/mvcc/lock.rs [hunk 2]
```diff
@@ -130,3 +130,67 @@ impl Lock {
...
+    fn test_lock() {
+        let mut locks = vec![Lock::new(LockType::Put, b"pk".to_vec(), 1, 10, None),
+                             Lock::new(LockType::Delete,
+                                       b"pk".to_vec(),
+                                       1,
+                                       10,
+                                       Some(b"short".to_vec()))];
+        for (i, lock) in locks.drain(..).enumerate() {
+            let v = lock.to_bytes();
+            match Lock::parse(&v[..]) {
```
Comment (siddontang) [inline src/storage/mvcc/lock.rs]:
```
let l = Lock::parse().unwrap_or_else(|e| {panic!()};
assert(l != lock, "#{}", i}
```

File: src/storage/mvcc/lock.rs [hunk 3]
```diff
@@ -130,3 +130,67 @@ impl Lock {
...
+    use super::*;
+
+    #[test]
+    fn test_lock_type() {
+        let (key, value) = (b"key", b"value");
+        let mut tests =
+            vec![(Mutation::Put((make_key(key), value.to_vec())), LockType::Put, FLAG_PUT),
+                 (Mutation::Delete(make_key(key)), LockType::Delete, FLAG_DELETE),
+                 (Mutation::Lock(make_key(key)), LockType::Lock, FLAG_LOCK)];
+        for (i, (mutation, lock_type, flag)) in tests.drain(..).enumerate() {
```
Comment (siddontang) [inline src/storage/mvcc/lock.rs]: no need using drain here.

**Comment by zhangjinpeng1987** [comment]
rest LGTM

File: src/storage/mvcc/lock.rs (hunk 3 above)
Comment (hhkbp2) [inline src/storage/mvcc/lock.rs]: It's fine to drain the vector here.

**Comment by hhkbp2** [comment]
PTAL @siddontang

File: src/storage/mvcc/lock.rs [hunk 4]
```diff
@@ -130,3 +130,59 @@ impl Lock {
...
+    fn test_lock() {
+        let mut locks = vec![Lock::new(LockType::Put, b"pk".to_vec(), 1, 10, None),
+                             Lock::new(LockType::Delete,
+                                       b"pk".to_vec(),
+                                       1,
+                                       10,
+                                       Some(b"short".to_vec()))];
+        for (i, lock) in locks.drain(..).enumerate() {
+            let v = lock.to_bytes();
+            let l = Lock::parse(&v[..]).unwrap_or_else(|e| panic!("#{} parse() err: {:?}", i, e));
```
Comment (AndreMouche) [inline src/storage/mvcc/lock.rs]: Should we add some tests when `parse` failed?

**Pull request merged**